android/src/main/java/io/ably/flutter/plugin/push/FirebaseMessagingReceiver.java

```diff
@@ -0,0 +1,89 @@
+package io.ably.flutter.plugin.push;
+
+import static io.ably.flutter.plugin.push.PushMessagingEventHandlers.PUSH_ON_BACKGROUND_MESSAGE_RECEIVED;
+import static io.ably.flutter.plugin.push.PushMessagingEventHandlers.PUSH_ON_MESSAGE_RECEIVED;
+
+import android.app.ActivityManager;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.util.Log;
+
+import androidx.annotation.Nullable;
+import androidx.localbroadcastmanager.content.LocalBroadcastManager;
+
+import com.google.firebase.messaging.RemoteMessage;
+
+import java.util.List;
+
+import io.ably.flutter.plugin.AblyFlutterPlugin;
+
+public class FirebaseMessagingReceiver extends BroadcastReceiver {
+  private static final String TAG = FirebaseMessagingReceiver.class.getName();
+
+  @Nullable
+  private static PendingResult asyncCompletionHandlerPendingResult = null;
```

Comment:
Hmm. Why is this a private static ? Looks confusing to me.

Reply:
It's not static anymore, thanks for catching that: 3d93df4513c4ca7127184bcddb9e6940c6286acc